web/src/compose_closed_ui.js

```diff
@@ -142,22 +160,37 @@ export function update_buttons_for_private() {
     update_buttons(text_stream, is_direct_message_narrow, disable_reply);
 }
 
-export function update_buttons_for_stream_views() {
+export function update_buttons_for_stream_views(filter) {
     const text_stream = $t({defaultMessage: "Start new conversation"});
+    const is_direct_message_narrow = false;
+    const stream_id = filter.has_operator("stream")
+        ? stream_data.get_stream_id(filter.operands("stream")[0])
+        : undefined;
+    const stream = sub_store.get(stream_id);
```

Comment:
Could some of this logic be replaced by having the caller pass in `narrow_state.stream_sub`?

Reply:
@timabbott, I've made the requested changes and updated the PR.